src/api/spec/controllers/webui/obs_factory/staging_projects_controller_spec.rb

```diff
@@ -121,33 +136,13 @@
   end
 
   describe 'GET #show' do
-    let(:source_package) { create(:package) }
-    let(:target_package) { create(:package, name: 'target_package', project: factory) }
-    let(:bs_request) do
-      create(:bs_request_with_submit_action,
-             target_project: factory.name,
-             target_package: target_package.name,
-             source_project: source_package.project.name,
-             source_package: source_package.name)
-    end
-    let(:description) do
-      <<-DESCRIPTION
-        requests:
-          - { id: #{bs_request.number} }
-      DESCRIPTION
-    end
-
     context 'with a existent factory_staging_project' do
-      before do
-        bs_request.update(state: 'declined')
-        factory_staging_a.update(description: description)
-      end
-
       context 'requesting html' do
-        subject { get :show, params: { project: factory, project_name: 'A' } }
+        subject! { get :show, params: { project: factory, project_name: 'A' } }
 
         it { is_expected.to have_http_status(:success) }
         it { expect(subject).to render_template(:show) }
+        it { expect(assigns(:staging_project).obsolete_requests).to contain_exactly(declined_bs_request) }
```

Comment:
couldn't something like `expect{ subject }.to change{ assigns(:staging_project).obsolete_requests }.to contain_exactly(declined_bs_request)` work? I find really ugly to define `subject` with `!`

Reply:
I don't think so. This would require that RSpec tracks any object instance that get's created during a test run (well, within this block).